src/app/clusters/icd-management-server/CodegenIntegration.cpp

```diff
@@ -0,0 +1,134 @@
+                                                   uint32_t optionalAttributeBits, uint32_t featureMap) override
+    {
+        ICDManagementCluster::OptionalAttributeSet optionalAttributeSet(optionalAttributeBits);
+        const BitMask<OptionalCommands> enabledCommands = kEnabledCommands();
+
+        // Get UserActiveModeTriggerHint
+        BitMask<IcdManagement::UserActiveModeTriggerBitmap> userActiveModeTriggerHint(0);
+        if (Clusters::IcdManagement::Attributes::UserActiveModeTriggerHint::Get(endpointId, &userActiveModeTriggerHint) !=
+            Protocols::InteractionModel::Status::Success)
+        {
+            ChipLogError(Zcl, "Failed to get UserActiveModeTriggerHint, using default (0)");
+        }
+
+        // Get UserActiveModeTriggerInstruction
+        char instructionBuffer[kUserActiveModeTriggerInstructionMaxLength];
+        MutableCharSpan instructionSpan(instructionBuffer, sizeof(instructionBuffer));
+        CharSpan userActiveModeTriggerInstruction;
+
+        if (Clusters::IcdManagement::Attributes::UserActiveModeTriggerInstruction::Get(endpointId, instructionSpan) !=
+            Protocols::InteractionModel::Status::Success)
+        {
+            ChipLogError(Zcl, "Failed to get UserActiveModeTriggerInstruction, using default (empty string)");
+            userActiveModeTriggerInstruction = CharSpan();
+        }
+        else
+        {
+            userActiveModeTriggerInstruction = CharSpan(instructionSpan.data(), instructionSpan.size());
+        }
+
+        gServer.Create(endpointId, Server::GetInstance().GetPersistentStorage(), *Server::GetInstance().GetSessionKeystore(),
+                       Server::GetInstance().GetFabricTable(), ICDConfigurationData::GetInstance().GetInstance(),
+                       optionalAttributeSet, enabledCommands, userActiveModeTriggerHint, userActiveModeTriggerInstruction);
+        return gServer.Registration();
+    }
+
+    ServerClusterInterface * FindRegistration(unsigned clusterInstanceIndex) override { return &gServer.Cluster(); }
+    void ReleaseRegistration(unsigned clusterInstanceIndex) override { gServer.Destroy(); }
+};
+} // namespace
+
+void MatterIcdManagementClusterInitCallback(EndpointId endpointId)
+{
+    IntegrationDelegate integrationDelegate;
+    CodegenClusterIntegration::RegisterServer(
+        {
+            .endpointId                = endpointId,
+            .clusterId                 = IcdManagement::Id,
+            .fixedClusterInstanceCount = static_cast<uint16_t>(IcdManagement::StaticApplicationConfig::kFixedClusterConfig.size()),
+            .maxClusterInstanceCount   = 1, // only root-node functionality supported by this implementation
+            .fetchFeatureMap           = false,
+            .fetchOptionalAttributes   = true,
+        },
+        integrationDelegate);
+
+#if CHIP_CONFIG_ENABLE_ICD_CIP
+    PersistentStorageDelegate & storage           = Server::GetInstance().GetPersistentStorage();
```

Comment:
We should use the storage in Startup/Shutdown rather than trying to access globals here.